Objects/abstract.c

```diff
@@ -2147,55 +2147,72 @@ PyMapping_HasKey(PyObject *o, PyObject *key)
     return 0;
 }
 
+static PyObject *
+method_output_as_list(PyObject *o, _Py_Identifier *meth_id)
+{
+    PyObject *it, *result, *meth_output;
+
+    assert(o != NULL);
+    meth_output = _PyObject_CallMethodId(o, meth_id, NULL);
+    if (meth_output == NULL || PyList_CheckExact(meth_output)) {
+        return meth_output;
+    }
+    it = PyObject_GetIter(meth_output);
+    Py_DECREF(meth_output);
+    if (it == NULL) {
+        if (PyErr_ExceptionMatches(PyExc_TypeError)) {
+            PyErr_Format(PyExc_TypeError,
+                         "%.200s.%s() must return an iterable",
+                         Py_TYPE(o)->tp_name,
+                         meth_id->string);
```

Comment:
Use `meth_id->string` and the %U format. The result of formatting is a PyUnicode object.

Reply:
I don't think i understand your advice..
`meth_id->string` is of type `const char*`, so when i use `%U`, it causes a crash.

Reply:
There is a redundant call of `PyUnicode_DecodeUTF8Stateful()` inside of `PyUnicode_FromFormatV` called from `PyErr_Format()`. It can be avoided by passing a PyUnicode object (`meth_id->string`) and using the %U format.

Reply:
IIUC, you suggest something like:
```
            PyErr_Format(PyExc_TypeError,
                         "%.200s.%U() returned a non-iterable (type %.200s)",
                         Py_TYPE(o)->tp_name,
                         PyUnicode_FromString(meth_id->string),
                         Py_TYPE(meth_output)->tp_name);
```
But then i would also have to handle refcounts, and the case in which `PyUnicode_FromString()` failed. Do you think it's worth it?

Reply:
No, I suggest to use `meth_id->string` which is a cached `PyUnicode_FromString(meth_id->string)`.

Reply:
Ah, I just noticed typos in my previous comments! Of course I meant `meth_id->object` rather of `meth_id->string`!